src/lib/kdf/prf_x942/prf_x942.cpp

```diff
@@ -20,48 +22,35 @@ namespace {
 * Encode an integer as an OCTET STRING
 */
 std::vector<uint8_t> encode_x942_int(uint32_t n) {
-   uint8_t n_buf[4] = {0};
-   store_be(n, n_buf);
+   const auto n_buf = store_be(n);
 
    std::vector<uint8_t> output;
-   DER_Encoder(output).encode(n_buf, 4, ASN1_Type::OctetString);
+   DER_Encoder(output).encode(n_buf.data(), n_buf.size(), ASN1_Type::OctetString);
    return output;
 }
 
 }  // namespace
 
-void X942_PRF::kdf(uint8_t key[],
-                   size_t key_len,
-                   const uint8_t secret[],
-                   size_t secret_len,
-                   const uint8_t salt[],
-                   size_t salt_len,
-                   const uint8_t label[],
-                   size_t label_len) const {
-   if(key_len == 0) {
+void X942_PRF::perform_kdf(std::span<uint8_t> key,
+                           std::span<const uint8_t> secret,
+                           std::span<const uint8_t> salt,
+                           std::span<const uint8_t> label) const {
+   if(key.empty()) {
       return;
    }
 
-   const size_t blocks_required = key_len / 20;  // Fixed to use SHA-1
-
-   if(blocks_required >= 0xFFFFFFFE) {
-      throw Invalid_Argument("X942_PRF maximum output length exceeeded");
-   }
+   constexpr size_t sha1_output_bytes = 20;  // Fixed to use SHA-1
+   const size_t blocks_required = key.size() / sha1_output_bytes;
+   BOTAN_ARG_CHECK(blocks_required < 0xFFFFFFFF, "X942_PRF maximum output length exceeeded");
```

Comment:
FE?

Reply:
32-bit counter, initialized at 1. We should (theoretically) be able to generate 2^32-1 unique blocks.